### ISSUE TITLE
What if sumfield misses an update due to some unknown error in the transaction.

### ISSUE THREAD

**Issue by jitendrapurohit** [progressivetech/net.ourpowerbase.sumfields#64]
For eg -

The custom fields added by this extension is configured to updated "Instantly".

![image](https://user-images.githubusercontent.com/5929648/65040430-04b54b00-d972-11e9-894c-557195502d73.png)

Seems if this update to the summary data is missed, it can never be updated by the extension unless we come to the settings page and re-save the values?

Can it be set to both? i.e, an instant update is required, but the backend cron job also checks if the data values are up-to-date.

We're finding a number of transactions in civi that are not updating sumfield values in some cases. Eg a contribution is getting added to civi, but the "Total" stat value in summary data remains as it is. I've tried to replicate it but haven't found any issues yet.

Currently, we're handling such cases by resaving the sumfield settings, which correctly updates all the custom field values.

**Comment by jmcclelland** [comment]
Yes - I agree, this is a problem. I'm not sure there is a reliable way to test to make sure all the values are accurate.

I wonder if we should have a (by default daily) cron job that simply re-creates from scratch?
